src/Database/Eloquent/Concerns/HasRelationships.php

```diff
@@ -63,21 +67,38 @@ public function hasOne($related, $foreignKey = null, $localKey = null)
 
         $localKey = $localKey ?: $this->getKeyName();
 
-        return new HasOne($instance->newQuery(), $this, $foreignKeys ?: $foreignKey, $localKey);
+        return $this->newHasOne($instance->newQuery(), $this, $foreignKeys ?: $foreignKey, $localKey);
+    }
+
+    /**
+     * Instantiate a new HasOne relationship.
+     *
+     * @param \Illuminate\Database\Eloquent\Builder $query
+     * @param \Illuminate\Database\Eloquent\Model   $parent
+     * @param string|array                          $foreignKey
+     * @param string|array                          $localKey
+     *
+     * @return \Awobaz\Compoships\Database\Eloquent\Relations\HasOne
+     */
+    protected function newHasOne(Builder $query, Model $parent, $foreignKey, $localKey)
```

Comment:
If these methods are the same as the parent, they shouldn't need to be included here.

Reply:
Reverted 2e22c0f.

Return type does not match to the parent one, they are required.